lib/js-file.js

```diff
@@ -12,24 +12,37 @@ var KEYWORD_OPERATORS = {
  * File representation for JSCS.
  *
  * @name JsFile
+ * @param {Object} options
+ * @param {String} options.filename
+ * @param {String} options.source
+ * @param {Object} options.esprima
+ * @param {Object} [options.esprimaOptions]
+ * @param {Boolean} [options.es3]
+ * @param {Boolean} [options.es6]
  */
-var JsFile = function(filename, source, tree, options) {
+var JsFile = function(options) {
```

Comment:
Most of the params are not optional. I introduces object because signature became to complicated. Probably the problem is the name: `options`. I can rename it to `params`.